### PR TITLE
R&R: Add basic add review modal functionality

### PR DESCRIPTION
**Pull request by davidguy3237**
Barebones add review modal is working. Still needs work, but it's functional at least.

### CODE REVIEW

File: client/src/components/ratings-and-reviews/RatingsAndReviews.jsx [hunk 1]
```diff
@@ -1,12 +1,42 @@
...
+  useEffect(() => {
+    if (product) {
+      axios.get('/reviews/meta', {
+        params: {
+          product_id: product.id,
+        },
+      })
+        .then(({ data }) => setMetaData(data));
+    }
+  }, [product]);
```
Comment (beanjermin) [inline client/src/components/ratings-and-reviews/RatingsAndReviews.jsx:21]: Don't forget to catch your errors

File: client/src/components/ratings-and-reviews/write-review/RateCharacteristic.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            id={`${descriptors[4]}-${charName}`}
+            name={charName}
+            value="5"
+            onClick={(e) => rateChar(charId, e.target.value)}
+          />
+        </label>
+      </span>
+    </div>
+  );
+}
```
Comment (beanjermin) [inline client/src/components/ratings-and-reviews/write-review/RateCharacteristic.jsx:84]: holy you did so much lol

File: client/src/components/ratings-and-reviews/write-review/WriteReviewModal.jsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                />
+              ))}
+            </div>
+            <button type="submit" id="write-review-submit">Submit Review</button>
+          </form>
+        </div>
+      </div>
+    </div>
+  );
+}
```
Comment (beanjermin) [inline client/src/components/ratings-and-reviews/write-review/WriteReviewModal.jsx:119]: omg my dude lol you got so much done

File: server/controllers/reviews.js [hunk 4]
```diff
@@ -30,13 +31,23 @@ module.exports = {
         authorization: API_TOKEN,
       },
     })
-      .then(({ data }) => res.json(data.ratings))
+      .then(({ data }) => res.json(data))
       .catch((err) => {
```
Comment (beanjermin) [inline server/controllers/reviews.js:35]: noted

File: server/controllers/reviews.js [hunk 5]
```diff
@@ -30,13 +31,23 @@ module.exports = {
...
+    axios.post(`${ATELIER_API}/reviews`, req.body, {
+      headers: {
+        authorization: API_TOKEN,
+      },
+    })
+      .then((results) => {
+        console.log('SUCCESS', results);
+        res.sendStatus(201);
+      });
+  },
```
Comment (beanjermin) [inline server/controllers/reviews.js:54]: missing a catch here too

Comment (davidguy3237) [reply server/controllers/reviews.js:54]: Added catch

File: client/src/components/ratings-and-reviews/RatingsAndReviews.jsx (hunk 1 above)
Comment (davidguy3237) [reply client/src/components/ratings-and-reviews/RatingsAndReviews.jsx:21]: added catch

**Review by beanjermin — APPROVED** [review]
Nice looks good to me

**Pull request merged**